doc/man3/SSL_shutdown.pod

```diff
@@ -104,11 +113,12 @@ and the peer's "close notify" alert was received.
 
 =item E<lt>0
 
-The shutdown was not successful because a fatal error occurred either
-at the protocol level or a connection failure occurred. It can also occur if
-action is need to continue the operation for non-blocking BIOs.
-Call L<SSL_get_error(3)> with the return value B<ret>
-to find out the reason.
+The shutdown was not successful.
+Call L<SSL_get_error(3)> with the return value B<ret> to find out the reason.
+It can occur if action is need to continue the operation for non-blocking BIOs.
+The output of L<SSL_get_error(3)> may be misleading, as an
+erroneous SSL_ERROR_SYSCALL may be flagged when not all data was read by
```

Comment:
I am not convinced that SSL_ERROR_SYSCALL is always non-fatal
and means that calling SSL_read is safe, when SSL_shutdown returns -1.

Reply:
So should I somewhere more explicitly say that you need to call SSL_read() before the 2nd SSL_shutdown()?

Reply:
But that would only be needed in the case if we're the one doing the shutdown. If the other peer did the shutdown and all data was already read, shutdown can return 1 on the first call.

Reply:
FYI: when we get the SSL_ERROR_SYSCALL in the following line:
https://github.com/openssl/openssl/blob/5585c8495e4c0175450b4ddc75b38eb76f929d34/ssl/record/rec_layer_s3.c#L1590
So I think the application data block is discarded, and the error is not like WANT_READ.

Reply:
I think the error here is not "erroneous" like in the case when SSL_shutdown returns 0.
The bidirectional shutdown by calling SSL_shutdown works only when both sides do it at the same time.

Reply:
I think there is a completely different shutdown sequence when one side calls SSL_shutdown and then
SSL_read in a loop until SSL_read returns ZERO_RETURN.
At that time the second SSL_shutdown is okay to do, but no longer necessary.

Reply:
I'm not sure what you're saying. I'm trying to rewrite the documentation on how I think it should work. If the code doesn't agree with that, the code should get fixed. I hope to have a new version of the manual soon.

Reply:
Are you planning on fixing the code, for this release?  If not then perhaps it's best to document current behavior and add a NOTE section.

Reply:
I have no idea what the current code does.

Reply:
I think it current returns an error when you try to read after SSL_shutdown(), which just seems wrong.

Reply:
Perhaps this should be discussed in email (project) list?  Rather than writing documentation that is known to be inaccurate?

Reply:
I tried read data after the first SSL_shutdown and it actually works.
The second SSL_shutdown can fail when application data is received.
I did not try to read after the error SSL_shutdown returns -1/SSL_ERROR_SYSCALL